synapse/storage/push_rule.py

```diff
@@ -185,6 +185,46 @@ def bulk_get_push_rules(self, user_ids):
 
         defer.returnValue(results)
 
+    @defer.inlineCallbacks
+    def copy_push_rules_from_room_to_room_for_user(self, old_room_id, new_room_id, user_id):
+        """Copy the push rules from one room to another for a specific user.
+
+        Args:
+            old_room_id (str): ID of the old room.
+            new_room_id (str): ID of the new room.
+            user_id (str): ID of user to copy push rules for.
+        """
+        # Retrieve push rules
+        user_push_rules = yield self.get_push_rules_for_user(user_id)
+
+        # Get rules for the old room
+        for rule in user_push_rules:
+            if old_room_id in rule["rule_id"]:
+                push_rules.append(rule)
```

Comment:
you haven't defined `push_rules`, but in any case: is there any point in a temporary list?

just do:

```
if not <rule applies to this room>:
    continue
<do the conversion>
```